### DISCUSSION TITLE
[Feature Request] API to Remove or Update Trace Tags

### DISCUSSION THREAD

**Discussion by 0xcodedead** [langfuse/langfuse#4301] [Ideas]
### Describe the feature or potential improvement

**Background**
When a trace tag is incorrectly specified, it can result in a large number of incorrect tags being applied. Manually removing these tags via the WebUI is a time-consuming process and can be error-prone.

Additionally, there are cases where it would be helpful to update specific parts of a completed trace, but currently, no functionality exists to address this need.

**Proposed Solution**
API to Remove Trace Tags

Provide an API endpoint that allows users to remove trace tags efficiently.
This would save time and reduce errors compared to manual deletion via the WebUI.
API to Update Completed Trace Data

Extend functionality to allow updates to specific parts of a completed trace.
This feature would enable more flexibility in managing and correcting trace data.

**Use Case**
A large set of traces is incorrectly tagged due to a misconfiguration.
The user needs to clean up the tags in bulk, which is impractical to do manually.
After cleanup, the user may also want to make targeted updates to trace data for accuracy or consistency.
Request
Would it be possible to implement APIs for:

Removing trace tags
Updating specific parts of completed traces
These features would greatly improve the manageability of trace data in complex use cases.

---

related to: https://github.com/orgs/langfuse/discussions/3289

**Comment by tilusnet** [comment]
Since this is unlikely to land soon (see langfuse/langfuse#8937, closed not planned), I built a Claude Code skill that works around it by always confirming a full proposed tag set with the user before applying anything — since a mistaken tag can't be walked back. Might be useful to others hitting this: https://github.com/tilusnet/langfuse-trace-tagging-skill
